plugins/woocommerce/src/Admin/Features/ProductBlockEditor/ProductTemplates/ProductTemplateInterface.php

```diff
@@ -0,0 +1,27 @@
+<?php
+/**
+ * Product Block Editor product template interface.
+ */
+
+namespace Automattic\WooCommerce\Admin\Features\ProductBlockEditor\ProductTemplates;
+
+/**
+ * Product Block Editor product template interface.
+ */
+interface ProductTemplateInterface {
```

Comment:
Do you foresee logic in these templates? It might be nice to have an easy JSON format for defining static templates.

Reply:
Yeah, that's what I'm thinking. I agree that it really just is a prettier representation of the PHP implementation. It does open the door for other editors to more easily handle it though, especially with a JSON schema.

A JSX representation would be interesting too... I guess I'm just desiring an easy-to-read declarative format for the template so that I can quickly visualize it while making changes.

Of course, down the road, an actual visual editor for templates would be super neat. But, I think that would be overkill given our current intended use of template. Actually though, it might not be that hard to implement, given the block editor's inherent support for visual editing.

> simply too much knowledge around currently registered items

Do you mean when you want to add a field/section?

I agree that for that case, an API would be far better.